easybuild/easyblocks/generic/bundle.py

```diff
@@ -120,15 +120,25 @@ def __init__(self, *args, **kwargs):
 
             # 'sources' is strictly required
             if cfg['sources']:
-                # add component sources to list of sources
-                self.cfg.update('sources', cfg['sources'])
+                # If per-component source URLs are provided, attach them directly to the relevant sources
+                if cfg['source_urls']:
+                    for source in cfg['sources']:
+                        if isinstance(source, basestring):
+                            self.cfg.update('sources', [{'filename': source, 'source_urls': cfg['source_urls']}])
+                        elif isinstance(source, dict):
+                            # Update source_urls in the 'source' dict to use the one for the components
+                            # (if it doesn't already exist)
+                            comp_source_urls = source.get('source_urls', cfg['source_urls'])
+                            self.cfg.update('sources', [source.update('source_urls', comp_source_urls)])
```

Comment:
This doesn't look correct...

The `.update` for `self.cfg` is something we've implemented ourselves (see `EasyConfig` class).

The `.update` of regular `dict` value (like `source` here) only accepts a `dict` value to update, and it doesn't return anything.

I think this is what you're intending:

```python
source['source_urls'] = comp_source_urls
self.cfg.update('sources', source)